ddpui/api/metrics_api.py

```
@@ -0,0 +1,326 @@
+
+
+@metrics_router.delete("/{metric_id}/")
+@has_permission(["can_edit_charts"])
+def delete_metric(request, metric_id: int):
+    """Delete a metric definition"""
+    org = request.orguser.org
+
+    try:
+        metric = MetricDefinition.objects.get(id=metric_id, org=org)
+    except MetricDefinition.DoesNotExist:
+        raise HttpError(404, "Metric not found")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)